lib/background/insight.js

```diff
@@ -1,33 +1,68 @@
-function initGoogleAnalytics(trackingCode, doTrack) {
-  if (!doTrack) {
-    return;
-  }
+/* eslint-disable id-length */
+
+import querystring from 'querystring';
+import pkg from '../../package.json';
+
+let instance;
+const baseUrl = 'https://ssl.google-analytics.com/collect';
 
-  /* eslint-disable */
-  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
-  /* eslint-enable */
+const baseParams = ({ tid, cid }) => ({
+  // Check out the measurement protocol parameter reference for details.
+  // https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+  v: 1,
+  tid, // Tracking ID
+  cid, // Anonymous identifier
+  aip: 1, // Anonymize IP,
+  an: pkg.name,
+  av: pkg.version,
+  ua: window.navigator.userAgent,
+  z: Date.now(),
+});
 
-  window.ga('create', 'UA-88792224-2', 'auto');
-  window.ga('set', 'anonymizeIp', true);
-  window.ga('set', 'checkProtocolTask', () => {}); // Removes failing protocol check. @see: http://stackoverflow.com/a/22152353/1958200
+function request(url, body) {
+  const xhr = new window.XMLHttpRequest();
+  xhr.open('POST', url);
+  xhr.send(querystring.stringify(body));
 }
 
-function trackEvent({ category, action, label, value }) {
-  if (!window.ga) {
-    return;
+const send = (requiredParams, params) => {
+  request(baseUrl, {
+    ...params,
+    ...baseParams(requiredParams),
```

Comment:
Is the intention here is for the `baseParams` to override the `params`? At first, I thought it might make more sense if the above two lines were reversed in order, but then I realized that I wasn't sure what the goal was.

For reference, here's the documentation I had to re-familiarize myself with: https://babeljs.io/docs/plugins/transform-object-rest-spread/